core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -865,25 +865,40 @@ public void deleteInode(RpcContext rpcContext, LockedInodePath inodePath, long o
     }
   }
 
+  private boolean checkPinningValidity(List<String> pinnedLocations) {
+    List<String> mediaList = ServerConfiguration.getList(PropertyKey.MASTER_TIERED_STORE_GLOBAL_MEDIA, ",");
+    for (String medium : pinnedLocations) {
+      if (!mediaList.remove(medium)) {
```

Comment:
I would prefer a contains check and having `pinnedLocations` be a set, unless you have a use case for duplicate medium types?